File: front/app/@types/Api/Cart/product.ts
import type { Categories } from ".";

interface Product {
  id: number,
  img: string,
  price: number,
  title: string,
  description: string,
  categories: Array<Categories>,
  type: string
  stok: number
}

interface Cart {
  id: number
  products: Array<CartProduct>
}

interface CartProduct {
  product: Product,
  quantity: number,
}

export type { Product, Cart, CartProduct }

File: api/entities/product.go
package entities

const (
	SPECIAL     = "especial"
	TRADITIONAL = "tradicional"
	ALCOHOLIC   = "alcoólico"
	WITH_MILK   = "com leite"
	ICE_COLD    = "gelado"
)

type Product struct {
	ID          int64    `json:"id"`
	Img         string   `json:"img"`
	Price       int64    `json:"price"`
	Title       string   `json:"title"`
	Description string   `json:"description"`
	Stok        int64    `json:"stok"`
	Categories  []string `json:"categories"`
	Type        string   `json:"type"`
}

type ProductSimple struct {
	ID          int64  `json:"id"`
	Img         string `json:"img"`
	Price       int64  `json:"price"`
	Title       string `json:"title"`
	Description string `json:"description"`
	Stok        int64  `json:"stok"`
	Categories  string `json:"categories"`
	Type        string `json:"type"`
}
